src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -1550,14 +1547,77 @@ public String toVariableNameString(WorldBorder border) {
 			.changer(DefaultChangers.entityChanger)
 		);
 
-
 		Classes.registerClass(new EnumClassInfo<>(EquipmentSlot.class, "equipmentslot", "equipment slots")
 			.user("equipment ?slots?")
 			.name("Equipment Slot")
 			.description("Represents an equipment slot of an entity.")
 			.since("2.11")
 		);
 
+		ClassInfo<?> pigVariantClassInfo = getRegsitryClassInfo(
+			"org.bukkit.entity.Pig$Variant",
+			"PIG_VARIANT",
+			"pigvariant",
+			"pig variants",
+			PigVariantDummy.class
+		);
+		Classes.registerClass(pigVariantClassInfo
+			.user("pig ?variants?")
+			.name("Pig Variant")
+			.description("Represents the variant of a pig entity.",
+				"NOTE: Minecraft namespaces are supported, ex: 'minecraft:warm'.")
+			.since("INSERT VERSION")
+			.requiredPlugins("Minecraft 1.21.5+")
+			.documentationId("PigVariant"));
+
+	}
+
+	/**
+	 * Gets a {@link RegistryClassInfo} by checking if the {@link Class} from {@code classPath} exists
+	 * 		and {@link Registry} or {@link io.papermc.paper.registry.RegistryKey} contains {@code registryName}.
+	 * If the {@link Class} from {@code classPath} does not exist or neither {@link Registry} and {@link io.papermc.paper.registry.RegistryKey}
+	 * 		does not contain {@code registryName}. A filler {@link ClassInfo} will be created using {@code dummyClass}.
+	 * @param classPath The {@link String} representation of the desired {@link Class}.
+	 * @param registryName The {@link String} representation of the desired {@link Registry}.
+	 * @param codeName The name used in patterns.
+	 * @param languageNode The language node of the type.
+	 * @param dummyClass The {@link Class} to be used for a filler {@link ClassInfo}.
+	 * @return {@link RegistryClassInfo} or a filler {@link ClassInfo}.
+	 */
+	private static <R extends Keyed> ClassInfo<?> getRegsitryClassInfo(
```

Comment:
I think i would rather see this method return null if it does not exist and let the caller decide what to do from there. A dummy class could be what's wanted, or perhaps no classinfo at all, so I think it's better to leave it up to the caller.